File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/drm/IDRM.java
package org.ngbp.jsonrpc4jtestharness.rpc.drm;

import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcMethod;
import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcType;

import org.ngbp.jsonrpc4jtestharness.rpc.drm.model.DRMOperation;
import org.ngbp.jsonrpc4jtestharness.rpc.drm.model.NotifyParams;

@JsonRpcType
public interface IDRM {
    @JsonRpcMethod("org.atsc.notify")
    NotifyParams drmNotification();

    @JsonRpcMethod("org.atsc.drmOperation")
    DRMOperation drmOperation();

}

File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/drm/IDRM.kt
package org.ngbp.jsonrpc4jtestharness.rpc.drm

import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcMethod
import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcType
import org.ngbp.jsonrpc4jtestharness.rpc.drm.model.DRMOperation

@JsonRpcType
interface IDRM {
    @JsonRpcMethod("org.atsc.drmOperation")
    fun drmOperation(): DRMOperation
}